### PR TITLE
Add recipe for iam-units

### PR DESCRIPTION
**Pull request by khaeru**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged
- [x] Source is from official source
- [x] Package does not vendor other packages.
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by khaeru** [comment]
@conda-forge/help-python this is ready to review/merge.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{% set name = "iam-units" %}
```
Comment (synapticarbors) [inline recipe/meta.yaml:1]: This file needs to be moved into `recipes/iam_units/meta.yaml`. Currently nothing is actually being built even though the tests all pass.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+  - python
+  - pip
```
Comment (synapticarbors) [inline recipe/meta.yaml:21]: It looks like you will probably need `setuptools_scm` as a `host` requirement

File: recipe/meta.yaml (hunk 1 above)
Comment (khaeru) [reply recipe/meta.yaml:1]: Hm, okay. I can do this.

I was looking at examples like [this](https://github.com/conda-forge/ixmp-feedstock/tree/master/recipe) and [https://github.com/conda-forge/xarray-feedstock/tree/master/recipe](this) where the meta.yaml file is in `recipes/` directly.

Is there a reason there is a difference, e.g. the layout has changed recently? Where should I look to find out about these recent changes, whether I need to apply them to other recipes, etc.?

File: recipe/meta.yaml (hunk 2 above)
Comment (khaeru) [reply recipe/meta.yaml]: Done.

**Comment by khaeru** [comment]
The following message appears in the CI log after the win_64 build and test succeed:
```
A subdirectory or file C:\\Miniconda\\conda-bld\\win-64\\ already exists.
##[error]Cmd.exe exited with code '1'.
```

I don't know what this means and don't have permissions to restart the build.

File: recipe/meta.yaml (hunk 1 above)
Comment (synapticarbors) [reply recipe/meta.yaml]: Nothing has changed. The layout in `staged-recipes` is different than the feedstocks. Here you should be looking at:

https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml

**Comment by synapticarbors** [comment]
Please re-merge `master` into this branch since a fix for the windows error has just been merged into `master`.

File: recipe/meta.yaml (hunk 1 above)
Comment (khaeru) [reply recipe/meta.yaml]: Thanks! It was the difference in layout that wasn't obvious.

**Comment by khaeru** [comment]
Rebased. Checks now all pass.

**Pull request merged**